airflow-core/src/airflow/api_fastapi/core_api/services/ui/grid.py

```diff
@@ -309,3 +311,204 @@ def _get_node_by_id(nodes, node_id):
         if node["id"] == node_id:
             return node
     return {}
+
+
+def fill_task_instance_summaries_simple(
+    grouped_task_instances: dict[str, list],
+    session: SessionDep,
+):
+    """
+    Fill the Task Instance Summaries for the Grouped Task Instances.
+
+    :param grouped_task_instances: Grouped Task Instances
+    :param task_instance_summaries_to_fill: Task Instance Summaries to fill
+    :param task_node_map: Task Node Map
+    :param session: Session
+
+    :return: None
+    """
+    # Additional logic to calculate the overall states to cascade recursive task states
+    # todo: must be a faster way
+    #   it also doesn't necessarily make sense if the last state in the group is success
```

Comment:
I wonder if we can just look at the leaf states? Because yeah, looking at everything isn't great - something in the middle could "fail", but be handled by a downstream via trigger rule...